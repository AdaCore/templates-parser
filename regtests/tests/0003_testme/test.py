from test_support import *

run('testme', ["testme2.tmplt"])
run('print_tree', ["testme2.tmplt"])
