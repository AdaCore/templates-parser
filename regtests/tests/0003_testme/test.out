

   one A Value
   two A Value
   three A Value

   ONE 
   TWO 
   THREE 

   atable
   with
   many
   values
   to
   be
   displayed
   one
   by
   one

  ok, VAR1 exist.

  ok, VAR1 exist.

  ok, VARx does not exist.

  ok, VARx does not exist.

  ok, VARx does not exist.

  ok, VARx does not exist.

  ok, VARx does not exist.

  ok, VAR1 exist.

A List : one, two, three
[INFO] testme2.tmplt
[C_INFO]  1 FALSE
[TEXT] 
[TABLE]
  [BLOCK]
    [COMMON]
      [TEXT]    @_VAR_DOES_NOT_EXIST_@
  [END_BLOCK]
[END_TABLE]
[TEXT] 
[TABLE]
  [BLOCK]
    [COMMON]
      [TEXT]    @_VAR3_@ @_CAPITALIZE:VAR1_@
  [END_BLOCK]
[END_TABLE]
[TEXT] 
[TABLE]
  [BLOCK]
    [COMMON]
      [TEXT]    @_UPPER:VAR3_@ @_VAR4_@
  [END_BLOCK]
[END_TABLE]
[TEXT] 
[TABLE]
  [BLOCK]
    [COMMON]
      [TEXT]    @_NO_SPACE:VAR2_@
  [END_BLOCK]
[END_TABLE]
[TEXT] 
[IF] @_EXIST:VAR1_@
  [TEXT]   ok, VAR1 exist.
[ELSE]
  [TEXT]   error, VAR1 exist.
[END_IF]
[TEXT] 
[IF] (@_EXIST:VAR1_@ = TRUE)
  [TEXT]   ok, VAR1 exist.
[ELSE]
  [TEXT]   error, VAR1 exist.
[END_IF]
[TEXT] 
[IF] @_EXIST:VARx_@
  [TEXT]   nok, VARx does not exist.
[ELSE]
  [TEXT]   ok, VARx does not exist.
[END_IF]
[TEXT] 
[IF] (@_EXIST:VARx_@ = TRUE)
  [TEXT]   nok, VARx does not exist.
[ELSE]
  [TEXT]   ok, VARx does not exist.
[END_IF]
[TEXT] 
[IF] (@_EXIST:VARx_@ = FALSE)
  [TEXT]   ok, VARx does not exist.
[ELSE]
  [TEXT]   nok, VARx does not exist.
[END_IF]
[TEXT] 
[IF] (@_IS_EMPTY:VARx_@ = TRUE)
  [TEXT]   ok, VARx does not exist.
[ELSE]
  [TEXT]   nok, VARx does not exist.
[END_IF]
[TEXT] 
[IF] @_IS_EMPTY:VARx_@
  [TEXT]   ok, VARx does not exist.
[ELSE]
  [TEXT]   nok, VARx does not exist.
[END_IF]
[TEXT] 
[IF] @_IS_EMPTY:VAR1_@
  [TEXT]   nok, VAR1 exist.
[ELSE]
  [TEXT]   ok, VAR1 exist.
[END_IF]
[TEXT] 
[TEXT] A List : @_VAR3_@
